subprojects/core/src/test/groovy/org/gradle/api/internal/artifacts/JavaEcosystemSupportTest.groovy

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts
+
+import org.gradle.api.attributes.Bundling
+import org.gradle.api.attributes.CompatibilityCheckDetails
+import org.gradle.api.attributes.CompileView
+import org.gradle.api.attributes.MultipleCandidatesDetails
+import org.gradle.api.attributes.Usage
+import org.gradle.internal.component.model.DefaultMultipleCandidateResult
+import org.gradle.util.TestUtil
+import spock.lang.Issue
+import spock.lang.Specification
+
+import static org.gradle.api.attributes.Bundling.EMBEDDED
+import static org.gradle.api.attributes.Bundling.EXTERNAL
+import static org.gradle.api.attributes.Bundling.SHADOWED
+
+class JavaEcosystemSupportTest extends Specification {
+    def "check usage compatibility rules (consumer value=#consumer, producer value=#producer, compatible=#compatible)"() {
+        given:
+        JavaEcosystemSupport.UsageCompatibilityRules rules = new JavaEcosystemSupport.UsageCompatibilityRules()
+        def details = Mock(CompatibilityCheckDetails)
+        when:
+        rules.execute(details)
+
+        then:
+        1 * details.getConsumerValue() >> usage(consumer)
+        1 * details.getProducerValue() >> usage(producer)
+        if (producer == consumer) {
+            // implementations are NOT required to say "compatible" because
+            // they should not even be called in this case
+            0 * details._()
+        } else if (compatible) {
+            1 * details.compatible()
+        } else {
+            0 * details._()
+        }
+
+        where:
+        consumer                     | producer                     | compatible
+        null                         | Usage.JAVA_API               | true
+        null                         | Usage.JAVA_RUNTIME           | true
+
+        Usage.JAVA_API               | Usage.JAVA_API               | true
+        Usage.JAVA_API               | Usage.JAVA_RUNTIME           | true
+
+        Usage.JAVA_RUNTIME           | Usage.JAVA_API               | false
+        Usage.JAVA_RUNTIME           | Usage.JAVA_RUNTIME           | true
+
+        // Temporary compatibility
```

Comment:
What is temporary compatibility?  Should this be a separate test marked with an `@Issue` or some note that it will go away?

Reply:
I did not write this, it was copied from the another test file where it didn't really belong. Though, looking at git history, it looks like some plugins still use these attribute values so we keep them around in our disambiguation rules: https://github.com/gradle/gradle/commit/81e8dfde3b07c09290b01fd9b4e50c6d698fb95a

This PR here seems to be the root of the change when these attribute values were deprecated: https://github.com/gradle/gradle/issues/9675

Though, I can't seem to find a long-term plan for how long we plan to keep these deprecated attribute values around. I worry that if we've published GMM with these attributes in the past, we can never truly remove them from our disambiguation rules.